tests/components/shelly/test_device_trigger.py

```diff
@@ -52,6 +57,71 @@ async def test_get_triggers(hass, coap_wrapper):
     assert_lists_same(triggers, expected_triggers)
 
 
+async def test_get_triggers_button(hass):
+    """Test we get the expected triggers from a shelly button."""
+    await async_setup_component(hass, "shelly", {})
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"sleep_period": 43200, "model": "SHBTN-1"},
+        unique_id="12345678",
+    )
+    config_entry.add_to_hass(hass)
+
+    device = Mock(
+        blocks=None,
+        settings=None,
+        shelly=None,
+        update=AsyncMock(),
+        initialized=False,
+    )
+
+    hass.data[DOMAIN] = {DATA_CONFIG_ENTRY: {}}
```

Comment:
We don't want to interact with integration details in the tests. We should just set up the integration and patch in the periphery, eg the library, as needed.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
This is not specific to this test, I am planning to rewrite all the tests, already tried some options but I was not happy with the result, If it is OK by you I would like to use your help for mocking the library in the right way.

Reply:
Sure!

Reply:
Looking at the library, one thing that makes the library harder to patch is that it is mixing the business logic of the `Device` class with the transport handling (aiohttp, coap and socket). If we'd break out the transport handling into a client class that the device can use, it would be easier to patch the client and know that all I/O is patched.